pkg/receive/handler.go

```diff
@@ -563,7 +563,7 @@ func (h *Handler) fanoutForward(pctx context.Context, tenant string, replicas ma
 			return fctx.Err()
 		case err, more := <-ec:
 			if !more {
-				return errs
+				return errs.Err()
```

Comment:
This is the thing that was causing 500 that should be 200

cc @squat @kakkoyun

Reply:
I thought tests were comprehensive enough to catch this :/